### ISSUE TITLE
Additional argument to constraint loss?

### ISSUE THREAD

**Issue by guoquan** [pylon-lib/pylon#13]
There are only `logits` in the input to constraint loss.
What is a proper way to add an additional argument to the loss in `Constraint`/`Solver`? For example `labels`, which is needed in IML?
